kore/src/Kore/Parser/LexerWrapper.hs

```diff
@@ -0,0 +1,166 @@
+runAlex fp input__ (Alex f) =
+    case f
+        ( AlexState
+            { alex_bpos = 0
+            , alex_pos = alexStartPos fp
+            , alex_inp = input__
+            , alex_chr = '\n'
+            , alex_scd = 0
+            }
+        ) of
+        Left msg -> Left msg
+        Right (_, a) -> Right a
+alexGetInput :: Alex AlexInput
+alexGetInput =
+    Alex $ \s@AlexState{alex_pos = pos, alex_bpos = bpos, alex_chr = c, alex_inp = inp__} ->
+        Right (s, AlexInput{alexPosn = pos, alexChar = c, alexStr = inp__, alexBytePos = bpos})
+
+alexSetInput :: AlexInput -> Alex ()
+alexSetInput AlexInput{alexPosn = pos, alexChar = c, alexStr = inp__, alexBytePos = bpos} =
+    Alex $ \s -> case s
+        { alex_pos = pos
+        , alex_bpos = bpos
+        , alex_chr = c
+        , alex_inp = inp__
+        } of
+        state__@(AlexState{}) -> Right (state__, ())
+
+alexGetStartCode :: Alex Int
+alexGetStartCode = Alex $ \s@AlexState{alex_scd = sc} -> Right (s, sc)
+
+alexSetStartCode :: Int -> Alex ()
+alexSetStartCode sc = Alex $ \s -> Right (s{alex_scd = sc}, ())
+
+alexEOF :: Alex Token
+alexEOF = do
+    AlexInput{alexPosn = p} <- alexGetInput
+    return $ Token p TokenEOF
+
+alexMonadScan = do
+    inp__@AlexInput{alexBytePos = n} <- alexGetInput
+    sc <- alexGetStartCode
+    case alexScanUser () inp__ sc of
+        AlexEOF -> alexEOF
+        AlexError (AlexInput{alexPosn = (AlexPn fp _ line column), alexStr = s}) ->
+            alexError fp line column $
+                if s == ""
+                    then "unexpected end of input"
+                    else "unexpected character " ++ show (ByteString.w2c $ ByteString.head s)
+        AlexSkip inp__' _len -> do
+            alexSetInput inp__'
+            alexMonadScan
+        AlexToken inp__'@AlexInput{alexBytePos = n'} _ action ->
+            let len = n' - n
+             in do
+                    alexSetInput inp__'
+                    action (ignorePendingBytes inp__) len
+
+-- -----------------------------------------------------------------------------
+-- Useful token actions
+
+type AlexAction result = AlexInput -> Int64 -> Alex result
+
+-- just ignore this token and scan another one
+-- skip :: AlexAction result
+skip _input _len = alexMonadScan
+
+-- ignore this token, but set the start code to a new value
+-- begin :: Int -> AlexAction result
+begin code _input _len = do alexSetStartCode code; alexMonadScan
+
+-- perform an action for this token, and set the start code to a new value
+andBegin :: AlexAction result -> Int -> AlexAction result
+(action `andBegin` code) input__ len = do
+    alexSetStartCode code
+    action input__ len
+
+token :: (AlexInput -> Int64 -> token) -> AlexAction token
+token t input__ len = return (t input__ len)
+
+-- -----------------------------------------------------------------------------
+-- Basic wrapper, ByteString version
+
+-- | Monad that repeats an operation until a boolean predicate returns True.
```

Comment:
Mention that this function is (transitively) only used for testing. If it should be more generally useful, it probably doesn't belong in this module and it could be optimized a bit.